File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> s(nums.begin(),nums.end());
        int longest = 0;
        for(int x: s){
            if(s.find(x-1)==s.end()){
                int len = 1;
                int current = x;
                while(s.find(current+1)!=s.end()){
                    len++;
                    current = current+1;
                }
                longest = max(longest, len);
            }
        }
        return longest;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        longest=0
        seen=set(nums)
        count=0

        for num in seen:
            if num-1 not in seen:
                count=1
                curr=num
                while(curr+1 in seen):
                    curr+=1
                    count+=1
                    
                longest=max(longest,count)

        return longest
        